frappe/desk/form/meta.py

```diff
@@ -48,9 +48,39 @@ def get_meta(doctype, cached=True) -> "FormMeta":
 		#       In prod don't use cached meta when explicitly requesting from DB.
 		meta = FormMeta(doctype, cached=frappe.conf.developer_mode)
 
+	if meta.name not in meta.special_doctypes:
+		meta = mask_protected_fields(meta)
+
+	return meta
+
+
+def mask_protected_fields(meta):
+	if (
+		frappe.flags.in_patch
+		or frappe.flags.in_install
+		or frappe.flags.in_migrate
+		or frappe.flags.in_setup_wizard
+	):
+		return meta
```

Comment:
This is a hack to avoid changing fieldtypes during migration. When you need to do something like this, realize that you're on a wrong path :skull: